services/brig/src/Brig/API/Public.hs

```diff
@@ -1134,11 +1141,23 @@ listConnections uid req = do
           -- Is this type actually useless? Or the tests not good enough?
           mtpPagingState = Public.ConnectionPagingState table (LBS.toStrict . C.unPagingState <$> pwsState)
         }
+
     mkState :: ByteString -> C.PagingState
     mkState = C.PagingState . LBS.fromStrict
 
-    convertedState :: Maybe C.PagingState
-    convertedState = fmap mkState . Public.mtpsState =<< Public.gmtprState req
+    localsAndRemotes :: Local UserId -> Maybe C.PagingState -> Range 1 500 Int32 -> Handler Public.ConnectionsPage
+    localsAndRemotes self pagingState size = do
```

Comment:
Is it worth trying to abstract out this logic, since it is the same for connections and conversations? In any case, let's do it in a separate PR.